### ISSUE TITLE
only address disappeared from wallet

### ISSUE THREAD

**Issue by brechindo** [iotaledger/android-wallet-app#148]
About 2 weeks ago, first installed, created first address, did first tx to it from binance.com
Tx appeared after about 48 hours, as Pending.
Stayed Pending for about another 48 hours.
Next morning, tx is gone because the address itself has vanished from the wallet.

Lost $4,300! Yeah yeah Im a fool for trusting a beta client in the first place. Regardless, any thoughts on how the address might be salvaged?  I've not reinstalled the app, not created any other addresses, so hopefully there's some deterministic PRNG magick to regen the one-and-only lost address?

Currently wallet 0.9.109-32eabfb installed from Play Store.
Currently Android 8.0.0 with 12/1/17 patch.  Rooted but not doing anything with root.
Google Nexus 6P

**Comment by hannesa2** [comment]
Yes, this app (or iota) has a huge network problem and I don't know if it's bad nodes, or the system 'iota' is bad. At least I reduced my engagement in iota because of this

But just whine is to less, what can be improved ?

The node selection with _separated_ protocol, ip, and port can be improved easily !
I see at least a predefined (working) node list **_in_** the app as a big step in the right direction. 
An a more advanced solution would be a user count who uses right now this node, or a rating for this node

**Comment by pinpong** [comment]
Sorry about this but the node is stuck with an older solid milestone. Let's wait for a bug fix. Andreas has already the log from the deadlock.

**Comment by hannesa2** [comment]
It's good to see someone takes care about this.
My positive benchmark is, when I can send a friend of my 1 IOTA within seconds, but currently IOTA has a failure of 100% on my side.
As I see, when the infrastructure fails, just by one node, the system fails.
That's bad ! 

What's about this: when the wallet recognize a non-synced node, it's iterate all known nodes till it's find a valid one.
Currently it's just suck, and that's not a valid option in my point of view
